pkg/io/jsonio/unmarshal.go

```diff
@@ -81,6 +80,7 @@ func validateSegments(jsonSegments spec.SpecSegments) error {
 	return nil
 }
 
+// read segments from conn-spec file
 func translateSegments(jsonSegments spec.SpecSegments, segmentType spec.Type) map[string][]string {
```

Comment:
Both comment and function name are somewhat misleading. Segments are not really read from file here, nor any translation takes place.
I suggest renaming this function to `filterSegmentsByType` or just `filterSegments`.